config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -32,59 +41,147 @@
+        final String uniqueNameWithPath = PathUtils.extractPath(nameWithPath) + uniqueName;
+        Element entityXml = null;
+        if(EntityTypes.POLICY_TYPE.equals(entity.getType())){
+            entityXml = getUniquePolicyXml(entity.getXml(), annotatedEntity, encassMap, projectName, projectVersion, uniqueName);
+        } else if(EntityTypes.ENCAPSULATED_ASSERTION_TYPE.equals(entity.getType())){
+            entityXml = (Element) entity.getXml().cloneNode(true);
+            final Element nameElement = getSingleChildElement(entityXml, NAME);
+            nameElement.setTextContent(uniqueName);
+        }  else {
+            entityXml = entity.getXml();
+        }
+        return EntityBuilderHelper.getEntityWithMappings(entity.getType(), uniqueNameWithPath, entity.getId(), entityXml, MappingActions.NEW_OR_UPDATE, entity.getMappingProperties());
     }
 
-    private Map<String, Pair<Element, BundleMetadata>> buildAnnotatedEntities(List<Entity> entities, Bundle bundle,
-                                                                              Document document, String projectName,
-                                                                              String projectGroupName,
-                                                                              String projectVersion) {
-        final Map<String, Pair<Element, BundleMetadata>> annotatedElements = new LinkedHashMap<>();
-
-        // Filter the bundle to export only annotated entities
-        // TODO : Enhance this logic to support services and policies
-        bundle.getEntities(Encass.class).values().stream()
-                .filter(Encass::hasAnnotated)
-                .map(encass -> createAnnotatedEntity(encass, projectName, projectVersion))
-                .forEach(annotatedEntity -> {
-                    if (annotatedEntity.isBundleTypeEnabled()) {
-                        // buildEncassDependencies
-                        List<Entity> entityList = getEntityDependencies(annotatedEntity.getPolicyName(), entities, bundle);
-                        LOGGER.log(Level.FINE, () -> "Entity list : " + entityList);
-                        // Create bundle
-                        final Element annotatedBundle = bundleDocumentBuilder.build(document, entityList);
-                        final BundleMetadata bundleMetadata = bundleMetadataBuilder.build(annotatedEntity, entityList
-                                , projectGroupName, projectVersion);
-                        annotatedElements.put(annotatedEntity.getBundleName(), ImmutablePair.of(annotatedBundle,
-                                bundleMetadata));
+    private Element getUniquePolicyXml(final Element entityXml, final AnnotatedEntity annotatedEntity, final Map<String, Encass> encassMap, final String projectName, final String projectVersion, final String uniqueName) {
+        Element policyElement = (Element) entityXml.cloneNode(true);
```

Comment:
you just use entityXml, as Entity.getXml gives clone Element every time.

Reply:
here it is element

Reply:
You are just making a clone of cloned `Element` why is that required?